deepchecks/vision/vision_data/batch_wrapper.py

```diff
@@ -0,0 +1,179 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021-2022 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Contains code for BatchWrapper."""
+from typing import Any, Callable, Dict, List, Sequence, TypeVar, Union, cast
+
+import numpy as np
+import torch
+
+from deepchecks.vision.utils.image_functions import crop_image
+from deepchecks.vision.utils.vision_properties import PropertiesInputType, calc_vision_properties, validate_properties
+from deepchecks.vision.vision_data import VisionData
+from deepchecks.vision.vision_data.utils import BatchOutputFormat, sequence_to_numpy
+
+__all__ = ['BatchWrapper']
+
+
+class BatchWrapper:
+    """Represents dataset batch returned by the dataloader during iteration."""
+
+    def __init__(
+            self,
+            batch: BatchOutputFormat,
+            vision_data: VisionData,
+    ):
+        self._vision_data = vision_data
+        self._batch = batch
+        self._labels, self._predictions, self._images = None, None, None
+        self._embeddings, self._additional_data, = None, None
+        self._image_identifiers = batch.get('image_identifiers')
+        # if there are no image identifiers, use the number of  the image in loading process as identifier
+        if self._image_identifiers is None:
+            images_seen = self._vision_data.number_of_images_cached
+            self._image_identifiers = np.asarray(range(images_seen, images_seen + self.__len__()), dtype='str')
+
+        self._vision_properties_cache = dict.fromkeys(PropertiesInputType)
+
+    def _get_cropped_images(self):
+        imgs = []
+        for img, labels in zip(self.images, self.labels):
```

Comment:
Don't think this code runs.

1. There is no self.labels, self.images.
2. label is numpy and label.cpu() will raise an error